exceptions/exceptions.hh

```diff
@@ -301,4 +301,9 @@ public:
     unsupported_operation_exception(const sstring& msg) : std::runtime_error("unsupported operation: " + msg) {}
 };
 
+struct mutation_unstarted_write_cancelled_exception : public std::runtime_error {
+public:
+    mutation_unstarted_write_cancelled_exception() : std::runtime_error("unstarted write cancelled") {}
+};
+
```

Comment:
Why is an unstarted write being canceled is more benign than a started write being canceled?

Reply:
I think a started write being cancelled is never more benign than the unstarted one, but it can definitely be less benign - e.g. if some side effects already happened on the write path.

This exception type is specified only because we actually *expect* it to happen during normal code path, because when another node shuts down, we time out pending mv writes for that node. Treating them as regular errors will result in terrifying users by showing `ERROR:` messages in the log and also confuses dtests.

Option 2 is to not treat it as an error at all, and I agree with that at the conceptual level, because I'm not a fan of exception-driven development. But, the current behaviour is that `unstarted write cancelled` situations result in exceptional futures that break the `then()` chain of sending the mutations further, and I think that it's dangerous to just convert them to regular futures and hope that everyting works fine. So, the proper way could be to never throw `mutation_unstarted_write_cancelled_exception` at all, but rather make sure that if an unstarted write is cancelled, there's a condition somewhere down the chain that checks if the write is still not cancelled and can proceed, but without actually propagating exceptional futures through the chain. I can try that approach later (Monday) and see if it's simple enough.